### ISSUE TITLE
v1.88.1 does not connect to older Linux distributions like v1.87.2 does

### ISSUE THREAD

**Issue by aschmalt** [microsoft/vscode#211718]
Type: <b>Bug</b>

Taken from: https://code.visualstudio.com/docs/remote/faq#_can-i-run-vs-code-server-on-older-linux-distributions 
`Starting with VS Code release 1.86.1 (January 2024), the minimum requirements for the build toolchain of the remote server were raised. The prebuilt servers distributed by VS Code are compatible with Linux distributions based on glibc 2.28 or later, for example, Debian 10, RHEL 8, or Ubuntu 20.04. VS Code will still allow users to connect to an OS that is not supported by VS Code (OS that does not provide glibc >= 2.28 and libstdc++ >= 3.4.25) until February 2025. This allows time for you and your companies to migrate to newer Linux distributions. VS Code will show a dialog and banner message when you connect to an OS version that is not supported by VS Code1`

The linux distribution connected to in v1.87.2 works fine, but in v1.88.1 (or v1.88.0) the connection does not work.   

VS Code version: Code 1.87.2 (863d2581ecda6849923a2118d93a088b0745d9d6, 2024-03-08T15:20:17.278Z)
OS version: Windows_NT x64 10.0.22631
Modes:
Remote OS version: Linux x64 4.12.14-122.136-default

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz (8 x 2995)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.69GB (3.12GB free)|
|Process Argv|--crash-reporter-id fac7a100-124d-477e-b91e-64da8f7d8173|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: scee432510.zsc9.intel.com|
|OS|Linux x64 4.12.14-122.136-default|
|CPUs|Intel(R) Xeon(R) Gold 6336Y CPU @ 2.40GHz (24 x 2400)|
|Memory (System)|1007.24GB (374.34GB free)|
|VM|0%|
</details><details><summary>Extensions (57)</summary>

Extension|Author (truncated)|Version
---|---|---
project-manager|ale|12.7.0
bracket-pair-colorizer-2|Coe|0.2.4
pythonsnippets|frh|1.0.2
remotehub|Git|0.62.0
vscode-peacock|joh|4.2.2
vsc-python-indent|Kev|1.18.0
vscode-edge-devtools|ms-|2.1.4
jupyter-keymap|ms-|1.1.2
remote-containers|ms-|0.338.1
remote-ssh|ms-|0.107.1
remote-ssh-edit|ms-|0.86.0
remote-wsl|ms-|0.86.0
vscode-remote-extensionpack|ms-|0.25.0
azure-repos|ms-|0.38.0
remote-explorer|ms-|0.4.1
remote-repositories|ms-|0.40.0
remote-server|ms-|1.5.0
emacs-mcx|tut|0.59.2
emacs|vsc|0.1.3
rainbow-brackets|2gu|0.0.6
project-manager|ale|12.7.0
asciidoctor-vscode|asc|3.1.10
gitignore|cod|0.9.0
vscode-markdownlint|Dav|0.54.0
git-extension-pack|don|0.1.3
githistory|don|0.6.20
gitlens|eam|14.7.0
vscode-github-actions|git|0.26.2
vscode-pull-request-github|Git|0.78.1
todo-tree|Gru|0.0.226
vscode-edit-csv|jan|0.8.3
jenkins-pipeline-linter-connector|jan|1.2.0
plantuml|jeb|2.18.1
svg|joc|1.5.3
restructuredtext|lex|189.3.0
rainbow-csv|mec|3.11.0
git-graph|mhu|1.30.0
autopep8|ms-|2023.8.0
pylint|ms-|2023.10.1
python|ms-|2024.0.1
vscode-pylance|ms-|2023.12.1
jupyter|ms-|2023.11.1100101639
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.17
vscode-jupyter-cell-tags|ms-|0.1.8
vscode-jupyter-slideshow|ms-|0.1.5
cmake-tools|ms-|1.16.32
cpptools|ms-|1.18.5
cpptools-extension-pack|ms-|1.3.0
makefile-tools|ms-|0.9.10
test-adapter-converter|ms-|0.1.9
autodocstring|njp|0.6.1
simple-rst|tro|1.5.4
cmake|twx|0.0.17
gitblame|wad|10.9.0
markdown-all-in-one|yzh|3.6.2
vscode-open-in-github|ziy|1.3.6

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscoreces:30445986
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
a9j8j154:30646983
962ge761:30959799
pythongtdpath:30769146
welcomedialog:30910333
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
2e7ec940:31000449
pythontbext0:30879054
accentitlementst:30995554
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
fegfb526:30981948
bg6jg535:30979843
ccp2r3:30993541
dsvsc020:30976470
pythonait:31006305
gee8j676:31009558
dsvsc021:30996838
bdiig495:31013172
pythoncenvpt:31022790

```

</details>

**Comment by kheuton** [comment]
I was able to connect with 1.88.0, but not upon upgrading to 1.88.1

**Comment by kheuton** [comment]
1.89.0 Also appears to be affected

**Comment by audetto** [comment]
There is definitely something different between 1.87 and 1.88+.
I have been moving up and down a lot, and unfortunately I cannot pinpoint to an exact issue.

But installing the vscode server on RHEL7 is a lot harder with 1.89 than it was before.

Probably a red herring, but downloading the vscode server is unreliable on 1.89 too.

At my last attempt it has worked after adding a http proxy (on Windows) which till 1.87 was always autodetected.
But then again, no password was requested, so I am not even sure it was used, nor that it is relevant.

**Comment by deepak1556** [comment]
If you are facing the issue with ssh extension, please make sure you are using `>= v0.110` and confirm if the issue persists. In other cases, please provide a minimal repro. Thanks!

**Comment by audetto** [comment]
I was using 0.112 and won't be able to provide any repro.
It happens in a network with authenticated http proxy, remote RHEL 7 on GCP without other network access.

1.88+ stands out since all previous updates went smoothly.
And even the glibc-related issue (of a few versions ago) was expected from the release notes.

Thank you anyway.

**Comment by deepak1556** [comment]
@audetto can you share the remote ssh logs

**Comment by tmchartrand** [comment]
I had this issue initially, and I believe it resolved after updating to remote-ssh v0.110.1.
My problem was I had pinned the extension version when the GLIBC issue first popped up, thinking that would keep it from installing a new vscode-server version on the remote. But in fact it seems to have done the opposite when paired with vscode versions 1.88/1.89!
Clearer documentation around all this would be great, it's still very hard to tell whether the intended support for remote access to older systems is only by accessing from vscode 1.85.2 (and perhaps accompanying extension versions) or whether newer (local) vscode and extension versions should be fine.

**Comment by audetto** [comment]
I have since deleted all local vscode folders on the remote and the latest update 1.89 to 1.89.1 was smooth.
